code/game/objects/items/reagent_containers/pill.dm

```diff
@@ -16,16 +16,16 @@ var/global/list/randomized_pill_icons
 	w_class = 1
 	volume = 60
 	var/pill_desc = "An unknown pill." //the real description of the pill, shown when examined by a medically trained person
+	var/pill_id
 
 /obj/item/reagent_container/pill/Initialize()
 	. = ..()
-	if(!randomized_pill_icons)
-		var/allowed_numbers = list(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21)
-		randomized_pill_icons = list()
-		for(var/i = 1 to 21)
-			randomized_pill_icons += "pill[pick_n_take(allowed_numbers)]"
+	if(!LAZYLEN(GLOB.randomized_pill_icons))
+		var/list/allowed_numbers = shuffle(list(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21))
+		for(var/i = 1 to allowed_numbers.len)
```

Comment:
```suggestion
		for(var/i in 1 to length(allowed_numbers))
```

Reply:
merely checking the .len var should suffice since the list should exist 1000% of the times.
using length() would be redunant.
Though I'll change that = to in

Reply:
.len is slower than length()

Reply:
Oh, boy.